File: app/src/main/java/com/example/pam_project/repositories/lists/ListsRepository.java
package com.example.pam_project.repositories.lists;

import com.example.pam_project.features.lists.list.ListInformation;

import io.reactivex.Completable;
import io.reactivex.Flowable;

public interface ListsRepository {

    ListInformation getList(final long id);

    Completable insertList(final String name, final long categoryId);

    Completable updateList(final long id, final String name, final long categoryId);

    Completable deleteList(final long id);

    Flowable<ListInformation> getListWithTasks(final long listId);

}

File: app/src/main/java/com/example/pam_project/repositories/lists/ListsRepository.kt
package com.example.pam_project.repositories.lists

import com.example.pam_project.features.lists.list.ListInformation
import io.reactivex.Completable
import io.reactivex.Flowable

interface ListsRepository {
    fun getList(id: Long): Flowable<ListInformation?>
    fun insertList(name: String, categoryId: Long): Completable
    fun updateList(id: Long, name: String, categoryId: Long): Completable
    fun deleteList(id: Long): Completable
    fun getListWithTasks(listId: Long): Flowable<ListInformation?>
}